src/models/LayoutSchema.ts

```diff
@@ -0,0 +1,42 @@
+export class LayoutSchema {
+    public static readonly INITIAL_LAYOUT_SCHEMA_VERSION = 1;
+    public static readonly CURRENT_LAYOUT_SCHEMA_VERSION = 1;
+
+    public static isLayoutVersionValid = (version: number): boolean => {
+        return Number.isInteger(version) && version >= LayoutSchema.INITIAL_LAYOUT_SCHEMA_VERSION && version <= LayoutSchema.CURRENT_LAYOUT_SCHEMA_VERSION;
+    };
+
+    // key: layout schema version, value: schema
+    public static readonly LAYOUT_SCHEMAS = {
+        "1" : {
+            "required": ["layoutVersion", "docked", "floating"],
+            "properties": {
+                "layoutVersion": {
+                    "type": "integer",
+                    "minimum": LayoutSchema.INITIAL_LAYOUT_SCHEMA_VERSION,
+                    "maximum": LayoutSchema.CURRENT_LAYOUT_SCHEMA_VERSION
+                },
+                "docked":  {
+                    "type": "object",
+                    "properties": {
+                        "type": {
+                            "type": "string"
+                        },
+                        "content": {
+                            "type": "array",
+                            "items": {
+                                "type": "object"
+                            }
+                        }
+                    }
+                },
+                "floating": {
+                    "type": "array",
+                    "items": {
```

Comment:
Do you think we should be stricter with the types of widgets we can store in the schema?

@kswang1029 Currently the schema only specifies layout properties, rather than properties of the widgets themself. Should we store things like plot colours, plot types, etc?

Reply:
trying to add the new widget properties to layout now